test/test_text_check.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: false
+
+module REXMLTests
+  class TextCheckTester < Test::Unit::TestCase
+    include REXML
+
+    def test_valid_pattern
+      chars = ['&A;', '&:;', '&_;', '&:A;', '&AA.bb-00;']
+      chars.each do |token|
+        text = Text.new(token, false, nil, true)
+        assert_equal(token, text.to_s)
+      end
```

Comment:
Could you create a test per pattern?

```ruby
def check(string)
  REXML::Text.check(string, REXML::Text::NEEDS_A_SECOND_CHECK, nil)
end

class TestValid < self
  class TestCharacterReference < self
    def test_unknown_ascii
      check("&A;")
    end

    def test_unknown_symbol
      check("&:;")
    end
    
    def test_defined_amp
      check("&amp;")
    end

    # ...
```

Reply:
I split the test cases at https://github.com/ruby/rexml/pull/165/commits/c2b376f405f1f9b7cce08c9adb2b1527a806dfd8 

And I reduced the test patterns.